app/src/main/java/com/github/se/eventradar/qrCode/QrCodeGenerator.kt

```diff
@@ -0,0 +1,82 @@
+package com.github.se.eventradar.qrCode
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.storage.FirebaseStorage
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.EncodeHintType
+import com.google.zxing.qrcode.QRCodeWriter
+import java.io.ByteArrayOutputStream
+import java.util.*
+
+class QrCodeGenerator {
+
+  fun qrCodeProcess(db: FirebaseFirestore, userId: String) {
```

Comment:
A better workflow would be:

1. call `qrCodeProcess`
2. `qrCodeProcess` generates the bitmap and uploads to cloud storage
3. `qrCodeProcess` _returns_ the download link
4. This download link is then used in the user object to upload the user to firebase

This way you don't need to worry about the userId. Now this is truly a utility function and doesn't depend on the user.